actiontext/lib/action_text/attachables/missing_attachable.rb

```diff
@@ -2,11 +2,25 @@
 
 module ActionText
   module Attachables
-    module MissingAttachable
+    class MissingAttachable
       extend ActiveModel::Naming
 
-      def self.to_partial_path
-        "action_text/attachables/missing_attachable"
+      DEFAULT_PARTIAL_PATH = "action_text/attachables/missing_attachable"
+
+      def initialize(sgid)
+        @sgid = SignedGlobalID.parse(sgid, for: ActionText::Attachable::LOCATOR_NAME)
+      end
+
+      def to_partial_path
+        if model
+          model.to_missing_attachable_partial_path
+        else
+          DEFAULT_PARTIAL_PATH
+        end
+      end
+
+      def model
+        @sgid&.model_name.to_s.safe_constantize
```

Comment:
I think you could use the `model_class` method here: 

```suggestion
        @sgid&.model_class
```

Reply:
Looking at the source for `globalid` I think this would raise in the case that there was actiontext content that referenced a class that was removed as it uses `constantize` vs `safe_constantize` internally.